File: frontend/src/services/api/calendar.ts
import api from './client'
import { CalendarEvent } from '../../types'

export interface CalendarEventListResponse {
  events: CalendarEvent[]
  total: number
}

export interface CalendarEventCreateRequest {
  title: string
  start_time: string
  end_time: string
  type: 'meeting' | 'deadline' | 'personal'
  is_private?: boolean
}

export interface CalendarEventUpdateRequest {
  title?: string
  start_time?: string
  end_time?: string
  type?: 'meeting' | 'deadline' | 'personal'
  is_private?: boolean
}

export const calendarService = {
  async getEvents(projectId: string, startDate?: string, endDate?: string): Promise<CalendarEventListResponse> {
    const params: Record<string, string> = {}
    if (startDate) params.start_date = startDate
    if (endDate) params.end_date = endDate
    const response = await api.get<CalendarEventListResponse>(
      `/calendar/projects/${projectId}`,
      { params }
    )
    return response.data
  },

  async createEvent(projectId: string, data: CalendarEventCreateRequest): Promise<CalendarEvent> {
    const response = await api.post<CalendarEvent>(
      `/calendar/projects/${projectId}`,
      data
    )
    return response.data
  },

  async updateEvent(eventId: string, data: CalendarEventUpdateRequest): Promise<CalendarEvent> {
    const response = await api.put<CalendarEvent>(
      `/calendar/${eventId}`,
      data
    )
    return response.data
  },

  async deleteEvent(eventId: string): Promise<void> {
    await api.delete(`/calendar/${eventId}`)
  },
}

File: backend/app/core/schemas/calendar.py
"""Calendar event schemas."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class CalendarEventResponse(BaseModel):
    """Calendar event response schema."""

    id: str
    project_id: str
    title: str
    start_time: str
    end_time: str
    type: str
    created_by: str
    is_private: bool
    created_at: str

    class Config:
        """Pydantic config."""

        from_attributes = True


class CalendarEventCreateRequest(BaseModel):
    """Calendar event create request schema."""

    title: str = Field(..., min_length=1, max_length=200)
    start_time: datetime
    end_time: datetime
    type: str = Field(default="meeting", pattern="^(meeting|deadline|personal)$")
    is_private: Optional[bool] = False


class CalendarEventUpdateRequest(BaseModel):
    """Calendar event update request schema."""

    title: Optional[str] = Field(None, min_length=1, max_length=200)
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    is_private: Optional[bool] = None


class CalendarEventListResponse(BaseModel):
    """Calendar event list response schema."""

    events: List[CalendarEventResponse]

